File: clients/python/amenolink/_http_requests.py
import json
import urllib.request
import urllib.error


def _post_json(url: str, data: dict) -> dict:
    json_bytes = json.dumps(data).encode('utf-8')
    request = urllib.request.Request(
        url=url,
        data=json_bytes,
        headers={'Content-Type': 'application/json'},
        method='POST',
    )
    with urllib.request.urlopen(request) as response:
        body = response.read().decode('utf-8')
        return json.loads(body) if body else {}

File: clients/dart/amenolink/lib/src/http_requests.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

Future<Map<String, dynamic>> postJson(String url, Map<String, dynamic> data) async {
  final response = await http.post(
    Uri.parse(url),
    headers: {'Content-Type': 'application/json'},
    body: jsonEncode(data),
  );

  if (response.body.trim().isEmpty) {
    return {};
  }

  return jsonDecode(response.body) as Map<String, dynamic>;
}
